### PR TITLE
Fixes #308 - don't mangle `stdin` and `stdout` overrides

### LINKED ISSUE #308: `stdin` and `stdout` cannot be `process.stdin` or `process.stdout`..?
hi there :wave:

thanks for the awesome lib :pray:

i ran into some odd behavior and i'm wondering what's up. using this script:

```js
#!/usr/bin/env node

const Enquirer = require('enquirer');


(async () => {
	console.log(':::: Running repro script....');
	try {
		const stdin = process.stdin;
		const stdout = process.stdout;
		const enquirer = new Enquirer({ stdin, stdout });
		const answer = await enquirer.prompt({
			type: 'input',
			name: 'test',
			message: 'Type something...'
		});
		console.log(answer);
	} catch (error){
		console.error(':::: ERROR:', error);
		return process.exit(1);
	}
})();
```

...i get the following output:

```
:::: Running repro script....
:::: ERROR: RangeError: Maximum call stack size exceeded
    at Function.keys (<anonymous>)
    at Object.exports.mixin (/Users/me/test/node_modules/enquirer/lib/utils.js:193:26)
    at Object.exports.merge (/Users/me/test/node_modules/enquirer/lib/utils.js:215:33)
    at Object.exports.mixin (/Users/me/test/node_modules/enquirer/lib/utils.js:199:33)
    at Object.exports.merge (/Users/me/test/node_modules/enquirer/lib/utils.js:215:33)
    at Object.exports.mixin (/Users/me/test/node_modules/enquirer/lib/utils.js:199:33)
    at Object.exports.merge (/Users/me/test/node_modules/enquirer/lib/utils.js:215:33)
    at Object.exports.mixin (/Users/me/test/node_modules/enquirer/lib/utils.js:199:33)
    at Object.exports.merge (/Users/me/test/node_modules/enquirer/lib/utils.js:215:33)
    at Object.exports.mixin (/Users/me/test/node_modules/enquirer/lib/utils.js:199:33)
```

when i delete `console.log(':::: Running repro script....');`, i get:

```
(node:13715) UnhandledPromiseRejectionWarning: TypeError: stream.listenerCount is not a function
    at emitKeypressEvents (readline.js:1165:14)
    at new Interface (readline.js:241:5)
    at Object.createInterface (readline.js:92:10)
    at Function.keypress.listen (/Users/me/test/node_modules/enquirer/lib/keypress.js:202:21)
    at Input.start (/Users/me/test/node_modules/enquirer/lib/prompt.js:180:28)
    at Input.initialize (/Users/me/test/node_modules/enquirer/lib/prompt.js:220:16)
    at /Users/me/test/node_modules/enquirer/lib/prompt.js:236:18
```

when i don't pass in `stdin` (e.g. `new Enquirer({ stdout });`), i get:

```
(node:13725) UnhandledPromiseRejectionWarning: TypeError: this.stdout.write is not a function
    at Input.cursorHide (/Users/me/test/node_modules/enquirer/lib/prompt.js:59:17)
    at module.exports (/Users/me/test/node_modules/enquirer/lib/placeholder.js:19:10)
    at Input.format (/Users/me/test/node_modules/enquirer/lib/types/string.js:156:14)
    at async Input.render (/Users/me/test/node_modules/enquirer/lib/types/string.js:172:18)
    at async Input.initialize (/Users/me/test/node_modules/enquirer/lib/prompt.js:221:5)
    at async /Users/me/test/node_modules/enquirer/lib/prompt.js:236:7
```

...and the process hangs until i press `CTRL+C`.

ultimately, i'm trying to wrap `enquirer` such that a series of ui helpers all use the same `stdin`, `stdout`, and `stderr` in order to make testing easier, etc.

my env info:
* macOS `v10.15.6`
* Node `v12.18.3`
* Enquirer `v2.3.6`

### PR DESCRIPTION
**Pull request by busticated**

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -195,7 +195,7 @@ exports.mixin = (target, b) => {
     if (desc.hasOwnProperty('value')) {
       if (target.hasOwnProperty(key) && isObject(desc.value)) {
         let existing = Object.getOwnPropertyDescriptor(target, key);
-        if (isObject(existing.value)) {
+        if (isObject(existing.value) && existing.value !== desc.value) {
```
Comment (jonschlinkert) [inline lib/utils.js:198]: what if `value` is Boolean `false`?

File: test/enquirer.js [hunk 2]
```diff
@@ -129,6 +129,36 @@ describe('Enquirer', function() {
...
+        name: 'test',
+        message: 'Type something?'
+      })
+      .then(answers => {
+        assert.equal(count, 1);
+        assert.equal(answers.test, 'ok');
+        cb(error);
+      });
+    });
+
```
Comment (jonschlinkert) [inline test/enquirer.js:161]: Thanks for doing this, looks great!

File: lib/utils.js (hunk 1 above)
Comment (busticated) [reply lib/utils.js:198]: then it'll fail the `isObject()` tests and never hit this line..?

i'm a bit fuzzy on what `mixin` needs to do as compared to something like `Object.assign()` tbh - happy to toss some tests at it if you have specific use-cases in mind.

**Comment by busticated** [comment]
to the CI failure:

> No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

you'll ultimately want to track down the root cause but in the meantime mocha has an `--exit` flag ([docs](https://mochajs.org/#command-line-usage)) which will force-quit tests upon completion that should help. happy to add that if you'd like :pray::+1:

File: lib/utils.js (hunk 1 above)
Comment (jonschlinkert) [reply lib/utils.js:198]: Sorry for the late reply. I remember making this comment on my phone, I must have been looking at something else before this. The code I thought I was commenting on did `if (!value)`. Sorry for the confusion. 

I think your code makes sense. I'll look into the CI failure.

**Comment by busticated** [comment]
@jonschlinkert @doowb hey there :wave: sorry to be _that_ guy but any chance we could get a bugfix release (`v2.3.7` or w/e) rolled out soon? getting this fix out into the world would really help unblock some things for me. as always, thanks for all the hard work here :pray::heart:

**Pull request merged**